rbx_binary/src/serializer.rs

```diff
@@ -623,6 +626,44 @@ impl<'a, W: Write> BinarySerializer<'a, W> {
                         chunk.write_interleaved_f32_array(x.into_iter())?;
                         chunk.write_interleaved_f32_array(y.into_iter())?;
                     }
+                    Type::CFrame => {
+                        let mut rotations = Vec::with_capacity(values.len());
+                        let mut x = Vec::with_capacity(values.len());
+                        let mut y = Vec::with_capacity(values.len());
+                        let mut z = Vec::with_capacity(values.len());
+
+                        for (i, rbx_value) in values {
+                            if let Variant::CFrame(value) = rbx_value.as_ref() {
+                                rotations.push(value.orientation);
+                                x.push(value.position.x);
+                                y.push(value.position.y);
+                                z.push(value.position.z);
+                            } else {
+                                return type_mismatch(i, &rbx_value, "CFrame");
+                            }
+                        }
+
+                        for matrix in rotations {
+                            //TODO handle special cases
```

Comment:
Can you file an issue and link it here? This comment is vague if the reader is unaware of the special forms that CFrames have. Maybe we should indicate that writing `0` here indicates the default CFrame representation.

Reply:
One might argue that people shouldn't be touching that code that writes CFrames if they're not familiar with the format! 😛

Linking it to an issue is a good idea regardless though, so I'll do that.